### PR TITLE
systemd: bring more in line with upstream

### PR DESCRIPTION
**Pull request by bb010g**
### Description

Robust escaping & automounts.

### Checklist

- [X] Change is backwards compatible.

- [X] Code formatted with `./format`.

- [X] Code tested through `nix-shell --pure tests -A run.all`.

- [X] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [X] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/doc/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [ ] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

  - [ ] Added myself and the module files to `.github/CODEOWNERS`.

### CODE REVIEW

File: modules/systemd.nix [hunk 1]
```diff
@@ -1,18 +1,24 @@
...
 
+with utils;
 with lib;
 
 let
 
   cfg = config.systemd.user;
 
+  # From <nixpkgs/nixos/modules/system/boot/systemd-lib.nix>
+  shellEscape = s: (replaceChars [ "\\" ] [ "\\\\" ] s);
```
Comment (rycee) [inline modules/systemd.nix:11]: Is this actually used?

File: modules/files.nix [hunk 2]
```diff
@@ -256,9 +256,9 @@ in
         }
         declare -A changedFiles
       '' + concatMapStrings (v: ''
-        _cmp "${sourceStorePath v}" "${homeDirectory}/${v.target}" \
-          && changedFiles["${v.target}"]=0 \
-          || changedFiles["${v.target}"]=1
+        _cmp --quiet ${escapeShellArg (sourceStorePath v)} ${escapeShellArg homeDirectory}/${escapeShellArg v.target} \
```
Comment (rycee) [inline modules/files.nix:259]: Hmm, I don't think `_cmp` takes a `--quiet` argument.

File: modules/systemd.nix [hunk 3]
```diff
@@ -1,18 +1,24 @@
-{ config, lib, pkgs, ... }:
+{ config, lib, pkgs, utils, ... }:
 
+with utils;
```
Comment (rycee) [inline modules/systemd.nix:3]: Which function from `utils` is used? I can't find any.

**Comment by rycee** [comment]
Thanks! Overall looks nice but I think it needs to be split into three different commits. One for the changes in files, one for the `automount`, and one for the other changes in `systemd.nix`.

File: modules/files.nix (hunk 2 above)
Comment (bb010g) [reply modules/files.nix:259]: Sorry, this was an improper merge.

File: modules/systemd.nix (hunk 3 above)
Comment (bb010g) [reply modules/systemd.nix:3]: Sorry, left over from removed code.

File: modules/systemd.nix (hunk 1 above)
Comment (bb010g) [reply modules/systemd.nix]: No; it was left over from removed code.

**Comment by sumnerevans** [comment]
Looks like this needs a rebase on the latest `master` so the tests pass.

**Comment by rycee** [comment]
Good work, merged to master!

**Pull request merged**